compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -301,4 +330,23 @@ object Completion {
     val Import: Mode = new Mode(4) | Term | Type
   }
 
+   /** A scope that tracks renames of the entered symbols.
+    *  Useful for providing completions for renamed symbols
+    *  in the REPL and the IDE.
+    */
+   private class RenameAwareScope extends Scopes.MutableScope {
+     private[this] val renames: mutable.Map[Symbol, Name] = mutable.Map.empty
```

Comment:
I don't think that's enough, the same symbol could be imported with multiple names:
```scala
import java.io.{File => JFile}
import java.io.{File => JFile2}
```